File: Day 57/MaximumUnitsOnaTruck.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
        
        sort(boxTypes.begin(), boxTypes.end(), [](const vector<int>& a, const vector<int>& b) {
            return a[1] > b[1];
        });

        int answer = 0;

        for(int i = 0 ; i < boxTypes.size() ; i++){

            int possible = min(truckSize, boxTypes[i][0]);

            answer += possible*boxTypes[i][1];

            truckSize -= possible;

            if(truckSize == 0) break;
        }

        return answer;




    }
};

int main(){
  
  return 0;
}

File: Day 57/Maximum Units on a Truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:

        sorted_vec = sorted(boxTypes, key=lambda x: x[1], reverse=True)

        answer = 0

        for i in range(0, len(boxTypes)):

            possible = min(truckSize, sorted_vec[i][0])

            answer += possible*sorted_vec[i][1]

            truckSize -= possible

            if(truckSize == 0):
                break
        

        return answer